search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -171,4 +162,44 @@ class TemporalEntityHandler(
         return buildGetSuccessResponse(mediaType, contextLink)
             .body(serializeObject(addContextsToEntity(temporalEntity, listOf(contextLink), mediaType)))
     }
+
+    /**
+     * Implements 5.6.15 - Delete Attribute instance from Temporal Representation of an Entity
+     */
+    @DeleteMapping("/{entityId}/attrs/{attrId}/{instanceId}")
+    suspend fun deleteAttributeInstanceTemporal(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @PathVariable entityId: String,
+        @PathVariable attrId: String,
+        @PathVariable instanceId: String
+    ): ResponseEntity<*> {
+        return either<APIException, ResponseEntity<*>> {
+            val sub = getSubFromSecurityContext()
+            val entityUri = entityId.toUri()
+            val instanceUri = instanceId.toUri()
+            val contexts = listOf(getContextFromLinkHeaderOrDefault(httpHeaders))
+            val expandedAttrId = JsonLdUtils.expandJsonLdTerm(attrId, contexts)!!
+
+            temporalEntityAttributeService.checkEntityExistence(entityUri).bind()
+
+            val temporalEntityAttributeUUID =
+                temporalEntityAttributeService.checkEntityAttributeExistence(entityUri, expandedAttrId).bind()
+
+            val canWriteEntity =
+                temporalEntityAttributeService.hasSpecificAccessPolicies(
+                    entityUri,
+                    listOf(SpecificAccessPolicy.AUTH_WRITE)
+                ).awaitFirst() ||
```

Comment:
you could move this business login into EntityAccessRightsService so that callers just have to call one canXXXEntity and not bother with authz details

Reply:
I move `hasSpecificAccessPolicies` into `EntityAccessRightsService` ?

Reply:
this check could be done transparently by the canXXXEntity method. As a caller, you don't care about the business rules, you just want to know if the current user can do the action.